okio/src/main/java/okio/Buffer.java

```diff
@@ -130,6 +139,26 @@ public long size() {
     };
   }
 
+  @Override public Reader reader() {
+    return new BomAwareReader(
+        null, BOM_UTF_8, BOM_UTF_32_LE, BOM_UTF_32_BE, BOM_UTF_16_LE, BOM_UTF_16_BE);
+  }
+
+  @Override public Reader reader(Charset charset) {
+    if (charset == null) throw new IllegalArgumentException("charset == null");
+
+    String charsetName = charset.name();
```

Comment:
What do you think about making BomAwareReader responsible for knowing which BOMs it can ignore for which charsets?